pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -94,19 +119,23 @@ func (np *NetworkPolicy) ruleConnections(rulePorts []netv1.NetworkPolicyPort, ds
 		if rulePorts[i].Protocol != nil {
 			protocol = *rulePorts[i].Protocol
 		}
-		ports := common.PortSet{}
+		ports := common.MakePortSet(false)
 		if rulePorts[i].Port == nil {
 			ports = common.MakePortSet(true)
 		} else {
-			startPort, endPort, err := np.getPortsRange(rulePorts[i].Port, rulePorts[i].EndPort, dst)
+			startPort, endPort, portName, err := np.getPortsRange(rulePorts[i].Port, rulePorts[i].EndPort, dst)
 			if err != nil {
 				return res, err
 			}
-			if isEmptyPortRange(startPort, endPort) {
-				continue
+			if dst == nil && portName != "" {
+				// adding namedPort to connectionSet in case of :
+				// -  dst is nil - for general connections;
+				// - TODO: if dst is a fake pod (representative)
```

Comment:
why is this a `todo` in the comment?

Reply:
i wanted to focus on general conns only in this PR (and test it locally);
so put this as TODO for next PR so I'll remeber to update and test

Reply:
The question was what is planned as a todo? Why does it require additional handling here?

Reply:
ah;
it serves the case of : a connection with named port when the dst peer is "representative";
in this case we can not convert the named port to a number; but still want to keep the "named port" in the connection as the potential port;

Reply:
but it seems to already be handled by the current code? how is it different from a real pod that does not have the specified named port?

Reply:
when a connection has "namedPort"; but the real pod does not actually has the named-port in its spec; this specified port/connection will be ignored and not saved as allowed connection between the peers.
if the real pod has the named-port in its spec; the number of the port will be appended to the allowed connection.

but in case of representative pod; the named-port may be a potential port for the connection